airflow/contrib/operators/gcp_compute_operator.py

```diff
@@ -0,0 +1,186 @@
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow import AirflowException
+from airflow.contrib.hooks.gcp_compute_hook import GceHook
+from airflow.contrib.utils.field_validator import BodyFieldValidator
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class GceInstanceStartOperator(BaseOperator):
+    """
+    Start an instance in Google Compute Engine.
+
+    :param project_id: Google Cloud Platform project where the Compute Engine
+        instance exists.
+    :type project_id: str
+    :param zone: Google Cloud Platform zone where the instance exists.
+    :type zone: str
+    :param resource_id: Name of the Compute Engine instance resource.
+    :type resource_id: str
+    :param gcp_conn_id: The connection ID used to connect to Google Cloud Platform.
+    :type gcp_conn_id: str
+    :param api_version: API version used (e.g. v1).
+    :type api_version: str
+    """
+    @apply_defaults
+    def __init__(self,
+                 project_id,
+                 zone,
+                 resource_id,
+                 gcp_conn_id='google_cloud_default',
+                 api_version='v1',
+                 *args, **kwargs):
+        self.project_id = project_id
+        self.zone = zone
+        self.full_location = 'projects/{}/zones/{}'.format(self.project_id,
+                                                           self.zone)
+        self.resource_id = resource_id
+        self.gcp_conn_id = gcp_conn_id
+        self.api_version = api_version
+        self._validate_inputs()
+        self._hook = GceHook(gcp_conn_id=self.gcp_conn_id, api_version=self.api_version)
+        super(GceInstanceStartOperator, self).__init__(*args, **kwargs)
+
+    def _validate_inputs(self):
+        if not self.project_id:
+            raise AirflowException("The required parameter 'project_id' is missing")
+        if not self.zone:
+            raise AirflowException("The required parameter 'zone' is missing")
+        if not self.resource_id:
+            raise AirflowException("The required parameter 'resource_id' is missing")
+
+    def execute(self, context):
+        return self._hook.start_instance(self.project_id, self.zone, self.resource_id)
+
+
+class GceInstanceStopOperator(BaseOperator):
+    """
+    Stop an instance in Google Compute Engine.
+
+    :param project_id: Google Cloud Platform project where the Compute Engine
+        instance exists.
+    :type project_id: str
+    :param zone: Google Cloud Platform zone where the instance exists.
+    :type zone: str
+    :param resource_id: Name of the Compute Engine instance resource.
+    :type resource_id: str
+    :param gcp_conn_id: The connection ID used to connect to Google Cloud Platform.
+    :type gcp_conn_id: str
+    :param api_version: API version used (e.g. v1).
+    :type api_version: str
+    """
+    @apply_defaults
+    def __init__(self,
```

Comment:
nit: It seems like you can extract a lot of common code between these into an abstract class. I'm not a Python expert though, so not sure if that's best practice

Reply:
Good call, after discussing it with @potiuk I've abstracted away `_validate_inputs()` and the constructor to an abstract `GceBaseOperator` (2d059a0e752b872c6db10e446b43364895bef877)

Reply:
+1